packages/plans-grid-next/src/hooks/data-store/use-grid-plans-for-comparison-grid.ts

```diff
@@ -0,0 +1,52 @@
+import {
+	FeatureList,
+	PLAN_ENTERPRISE_GRID_WPCOM,
+	PLAN_HOSTING_TRIAL_MONTHLY,
+} from '@automattic/calypso-products';
+import { useMemo } from 'react';
+import { GridPlan, PlansIntent } from '../../types';
+import useRestructuredPlanFeaturesForComparisonGrid from './use-restructured-plan-features-for-comparison-grid';
+
+interface Params {
+	allFeaturesList: FeatureList;
+	gridPlans: Omit< GridPlan, 'features' >[];
+	intent?: PlansIntent;
+	selectedFeature?: string | null;
+	showLegacyStorageFeature?: boolean;
+}
+
+const useGridPlansForComparisonGrid = ( {
+	allFeaturesList,
+	gridPlans,
+	intent,
+	selectedFeature,
+	showLegacyStorageFeature,
+}: Params ): GridPlan[] => {
+	const planFeaturesForComparisonGrid = useRestructuredPlanFeaturesForComparisonGrid( {
+		allFeaturesList,
+		gridPlans,
+		intent,
+		selectedFeature,
+		showLegacyStorageFeature,
+	} );
+
+	return useMemo( () => {
+		const hiddenPlans = [ PLAN_HOSTING_TRIAL_MONTHLY, PLAN_ENTERPRISE_GRID_WPCOM ];
```

Comment:
I wonder if we can find a way to consolidate this list of "hidden plans" with all of the "hidden plan flags" being refactored in https://github.com/Automattic/wp-calypso/pull/87478/files#diff-18214fd84316736ff6662f03fa201b6255f4c20f277c0eb683b5e36e9c9dea6dR430-R435 in a more central structure/abstraction.

My thinking is to avoid all of the individual props passed through like done in the follow-up and these other ones here. Perhaps a `hiddenPlans` array/object passed through instead that would include both?

that's also the intention with the TODO comment in the code:

> // TODO: `useFilterPlansForPlanFeatures` should gradually deprecate and whatever remains to fall into the `useGridPlans` hook

Reply:
Obviously something for the follow-up PR https://github.com/Automattic/wp-calypso/pull/87478

Reply:
> My thinking is to avoid all of the individual props passed through

I agree, too many props. I'll take a closer look in the followup 👍